src/mam4xx/ndrop.hpp

```diff
@@ -130,6 +245,1506 @@ void maxsat(
+                state_q, air_density, vaerosol[imode], qcldbrn1d_num[imode],
+                naerosol[imode]);
+
+  } // end imode
+} // loadaer
+
+KOKKOS_INLINE_FUNCTION
+void ccncalc(const Real state_q[nvars], const Real tair,
+             const Real qcldbrn[maxd_aspectype][AeroConfig::num_modes()],
+             const Real qcldbrn_num[AeroConfig::num_modes()],
+             const Real air_density,
+             const int lspectype_amode[maxd_aspectype][AeroConfig::num_modes()],
+             const Real specdens_amode[maxd_aspectype],
+             const Real spechygro[maxd_aspectype],
+             const int lmassptr_amode[maxd_aspectype][AeroConfig::num_modes()],
+             const Real voltonumbhi_amode[AeroConfig::num_modes()],
+             const Real voltonumblo_amode[AeroConfig::num_modes()],
+             const int numptr_amode[AeroConfig::num_modes()],
+             const int nspec_amode[AeroConfig::num_modes()],
+             const Real exp45logsig[AeroConfig::num_modes()],
+             const Real alogsig[AeroConfig::num_modes()], Real ccn[psat]) {
+
+  // calculates number concentration of aerosols activated as CCN at
+  // supersaturation supersat.
+  // assumes an internal mixture of multiple externally-mixed aerosol modes
+  // NOTE: CGS units
+
+  // Ghan et al., Atmos. Res., 1993, 198-221.
+
+  // input arguments
+  // @param [in] state_q(:,:,:) aerosol mmrs [kg/kg]
+  // @param [in] tair(:,:)     air temperature [K]
+  // @param [in] qcldbrn(:,:,:,:), qcldbrn_num(:,:,:) cloud-borne aerosol mass
+  // / number  mixing ratios [kg/kg or #/kg]
+  // @param [in] air_density(pcols,pver)       air density [kg/m3]
+
+  // output arguments
+  // ccn(pcols,pver,psat) number conc of aerosols activated at supersat [#/m3]
+  // qcldbrn  1) icol 2) ispec 3) kk 4)imode
+  // state_q 1) icol 2) kk 3) imode and ispec
+  const Real zero = 0;
+  // BAD CONSTANT
+  const Real nconc_thresh = 1.0e-3;
+  const Real twothird = 2.0 / 3.0;
+  const Real two = 2.0;
+  const Real three_fourths = 3.0 / 4.0;
+  const Real half = 0.5;
+  const Real one = 1;
+
+  const Real per_m3_to_per_cm3 = 1.0e-6;
+  //// const Real percent_to_fraction = 0.01;
+  // super(:)=supersat(:)*percent_to_fraction
+  // supersaturation [fraction]
+  const Real sq2 = haero::sqrt(2.0);
+  const Real two_over_root27 = 2.0 / haero::sqrt(27.0);
+  // BAD CONSTANT
+  // supersaturation (%) to determine ccn concentration
+  const Real super[psat] = {0.0002, 0.0005, 0.001, 0.002, 0.005, 0.01};
+  // phase of aerosol
+  const int phase = 3; // interstitial + cloudborne
+  const int nmodes = AeroConfig::num_modes();
+
+  const Real mwh2o = haero::Constants::molec_weight_h2o * 1e3; // [kg/kmol]
+  const Real r_universal = haero::Constants::r_gas * 1e3;      // [J/K/kmol]
+  const Real rhoh2o = haero::Constants::density_h2o;
+  const Real pi = haero::Constants::pi;
+
+  // surface tension coefficient [m-K] (from mam4)
+  // FIXME: any thoughts on what [m-K] means?
```

Comment:
In my experience, fit coefficients often can't be interpreted physically, at least not directly. It seems to me that `ssat_coeff` captures a power-law temperature dependence which is expressed in the form of this "surface tension" component. Note also the headache-inducing fractional-power (power law) units of `ssat_coeff`.